packages/material-ui/src/Slider/Slider.test.js

```diff
@@ -0,0 +1,470 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { spy, stub } from 'sinon';
+import { expect } from 'chai';
+import { createMount, getClasses } from '@material-ui/core/test-utils';
+import describeConformance from '@material-ui/core/test-utils/describeConformance';
+import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles';
+import { cleanup, createClientRender, fireEvent } from 'test/utils/createClientRender';
+import consoleErrorMock from 'test/utils/consoleErrorMock';
+import Slider from './Slider';
+
+function touchList(touchArray) {
+  touchArray.item = idx => touchArray[idx];
+  return touchArray;
+}
+
+function fireBodyMouseEvent(name, properties = {}) {
+  const event = document.createEvent('MouseEvents');
+  event.initEvent(name, true, true);
+  Object.keys(properties).forEach(key => {
+    event[key] = properties[key];
+  });
+  document.body.dispatchEvent(event);
+  return event;
+}
+
+describe('<Slider />', () => {
+  let mount;
+  let render;
+  let classes;
+
+  before(() => {
+    render = createClientRender({ strict: true });
+    classes = getClasses(<Slider value={0} />);
+    mount = createMount({ strict: true });
+  });
+
+  after(() => {
+    cleanup();
+    mount.cleanUp();
+  });
+
+  describeConformance(<Slider value={0} />, () => ({
+    classes,
+    inheritComponent: 'span',
+    mount,
+    refInstanceof: window.HTMLSpanElement,
+    testComponentPropWith: 'span',
+  }));
+
+  it('should call handlers', () => {
+    const handleChange = spy();
+    const handleChangeCommitted = spy();
+
+    const { container, getByRole } = render(
+      <Slider onChange={handleChange} onChangeCommitted={handleChangeCommitted} value={0} />,
+    );
+
+    fireEvent.mouseDown(container.firstChild);
+    document.body.dispatchEvent(new window.MouseEvent('mouseup'));
+
+    expect(handleChange.callCount).to.equal(1);
+    expect(handleChangeCommitted.callCount).to.equal(1);
+
+    fireEvent.keyDown(getByRole('slider'), {
+      key: 'Home',
+    });
+    expect(handleChange.callCount).to.equal(2);
+    expect(handleChangeCommitted.callCount).to.equal(2);
+  });
+
+  it('should only listen to changes from the same touchpoint', () => {
+    const handleChange = spy();
+    const handleChangeCommitted = spy();
+    const { container } = render(
+      <Slider onChange={handleChange} onChangeCommitted={handleChangeCommitted} value={0} />,
+    );
+
+    const event = fireBodyMouseEvent('touchstart', {
```

Comment:
Test should use actual TouchEvents and be skipped in browsers without them (edge, IE11)

Reply:
Does it matter?

Reply:
What happened between you arguing for browser test and this statement? You can't just flip-flop whenever you feel like it.

Reply:
The tests seem to pass in the browser, does it mean it's a valid usage (even if not ideal)?

Reply:
However, I agree with your point regarding the fact that the simulated interactions we produce should be as close as possible to what happens with a user.